src/d3-cloud.controller.js

```diff
@@ -8,47 +8,43 @@
   d3CloudController.$inject = ['$scope'];
 
   function d3CloudController($scope) {
-    $scope.$watch('words', function(newValue, oldValue) {
-      var i = [];
-      var items = [];
-      var ignore = $scope.ignoreList;
-      var updateflag = 0;
-
-      if (newValue) {
-        for (i = 0; i < newValue.length; i++) {
-          if (ignore.indexOf(newValue[i].name) < 1) {
-            items.push(newValue[i]);
-          }
-        }
-
-        if ($scope.cloud) {
-          if (oldValue) {
-            if (oldValue.length !== newValue.length) {
-              updateflag = 1;
-            } else {
-              for (i = 0; i < newValue.length; i++) {
-                if (!updateflag & newValue[i].name !== oldValue[i].name & newValue[i].score !== oldValue[i].score) {
-                  updateflag = 1;
+    $scope.$watch('words', function (newValue, oldValue) {
+      angular.element(document).ready(function () {
+        var i = [];
+        var items = [];
+        var updateflag = 0;
+
+        if (newValue) {
+          items = $scope.filterFunction($scope.words);
```

Comment:
Additionally, I think it would be nice if the filter function would be called for each word/item separately. It would take the burden away from people to iterate over the items themselves to filter them, and just focus on one item at a time. So basically restore the old code, but replace the `if (ignore.indexOf(newValue[i].name) < 1)` with `if (filter(newValue[i]))`..